Multisig/Logic/Models/PrivateKey/web3swift/BIP32HDNode.swift

```diff
@@ -0,0 +1,303 @@
+//
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import BigInt
+import CryptoSwift
+
+// https://github.com/matter-labs/web3swift
+
+extension UInt32 {
+    public func serialize32() -> Data {
+        var bigEndian = self.bigEndian
+        let count = MemoryLayout<UInt32>.size
+        let bytePtr = withUnsafePointer(to: &bigEndian) {
+            $0.withMemoryRebound(to: UInt8.self, capacity: count) {
+                UnsafeBufferPointer(start: $0, count: count)
+            }
+        }
+        let byteArray = Array(bytePtr)
+        return Data(byteArray)
+    }
+}
+
+public class HDNode {
+    public struct HDversion{
+        public var privatePrefix: Data = Data(hex: "0x0488ADE4")
+        public var publicPrefix: Data = Data(hex: "0x0488B21E")
+        public init() {
+
+        }
+    }
+    public var path: String? = "m"
+    public var privateKey: Data? = nil
+    public var publicKey: Data
+    public var chaincode: Data
+    public var depth: UInt8
+    public var parentFingerprint: Data = Data(repeating: 0, count: 4)
+    public var childNumber: UInt32 = UInt32(0)
+    public var isHardened:Bool {
+        get {
+            return self.childNumber >= (UInt32(1) << 31)
+        }
+    }
+    public var index: UInt32 {
+        get {
+            if self.isHardened {
+                return self.childNumber - (UInt32(1) << 31)
+            } else {
+                return self.childNumber
+            }
+        }
+    }
+    public var hasPrivate:Bool {
+        get {
+            return privateKey != nil
+        }
+    }
+
+    init() {
+        publicKey = Data()
+        chaincode = Data()
+        depth = UInt8(0)
+    }
+
+    public convenience init?(_ serializedString: String) {
+        let data = Data(Base58.bytesFromBase58(serializedString))
+        self.init(data)
+    }
+
+    public init?(_ data: Data) {
+        guard data.count == 82 else {return nil}
+        let header = data[0..<4]
+        var serializePrivate = false
+        if header == HDNode.HDversion().privatePrefix {
+            serializePrivate = true
+        }
+        depth = data[4..<5].bytes[0]
+        parentFingerprint = data[5..<9]
+        let cNum = data[9..<13].bytes
+        childNumber = UnsafePointer(cNum).withMemoryRebound(to: UInt32.self, capacity: 1) {
+            $0.pointee
+        }
+        chaincode = data[13..<45]
+        if serializePrivate {
+            privateKey = data[46..<78]
+            guard let pubKey = HDNode.privateToPublic(privateKey!, compressed: true) else {return nil}
+            if pubKey[0] != 0x02 && pubKey[0] != 0x03 {return nil}
```

Comment:
please put spaces before and after return nil

Reply:
will do in the next PR